src/vs/workbench/contrib/chat/common/chatModel.ts

```diff
@@ -522,23 +532,44 @@ export class Response extends AbstractResponse implements IDisposable {
 			this._updateRepr(quiet);
 		} else if (progress.kind === 'textEdit' || progress.kind === 'notebookEdit') {
 			// merge edits for the same file no matter when they come in
-			const groupKind = progress.kind === 'textEdit' ? 'textEditGroup' : 'notebookEditGroup';
+			const notebookUri = CellUri.parse(progress.uri)?.notebook;
+			const uri = notebookUri ?? progress.uri;
+			const groupKind = progress.kind === 'textEdit' && !notebookUri ? 'textEditGroup' : 'notebookEditGroup';
 			let found = false;
 			for (let i = 0; !found && i < this._responseParts.length; i++) {
 				const candidate = this._responseParts[i];
-				if (candidate.kind === groupKind && !candidate.done && isEqual(candidate.uri, progress.uri)) {
-					candidate.edits.push(progress.edits as any);
+				if (candidate.kind === groupKind && !candidate.done && isEqual(candidate.uri, uri)) {
+					if (candidate.kind === 'textEditGroup') {
+						candidate.edits.push(progress.edits as any);
+					} else if (candidate.kind === 'notebookEditGroup') {
```

Comment:
You also need to check `!candidate.done` here because any edits added to an already-completed group will get lost.

I think you also need to do the same check on its URI in case multiple notebooks are being edited?